### DISCUSSION TITLE
cpu usage keep high during query vectors from indexed collection after restart milvus

### DISCUSSION THREAD

**Discussion by hawkphantomnet** [milvus-io/milvus#4390] [Q&A and General discussion] [answered]
I have created IVFLAT index for a collection with 50-dim vectors. The milvus instances is a container with 24 cores CPU.
After index finished, I perform searching vector and getting response in ~5ms average. The cpu usage of the milvus process was ~240%. 
Then I restarted the container and perform searching again. The response time was  ~6ms average, but the cpu usage of the milvus process raised to ~1700%.

Just wonder what happened that using a lot of CPU during the searching after milvus restarted? 
I think the restarted milvus was still using the index since the response time didn't change dramatically. I tested searching without index, which response time was changed to ~17ms.

Milvus version: 0.10.4 
Vector dimension: 50 
Vector number: 280w 
Index type: IVFLAT
Metric type: L2

logs: 
before restarting milvus(cpu usage 240%): 
[has_index_low_cpu.zip](https://github.com/milvus-io/milvus/files/5649142/has_index_low_cpu.zip)

after restarting milvus(cpu usage 1700%):
[has_index_high_cpu.zip](https://github.com/milvus-io/milvus/files/5649144/has_index_high_cpu.zip)

**Comment by shiyu22** [comment]
Does this problem occur with every restart? Since your queries are frequent and therefore have a high CPU footprint, it may be that there are other tasks taking up space on the first query.

**Reply by hawkphantomnet** [reply]
> Does this problem occur with every restart? 
yes

And if I recreate the index for the segments, the later queries will cost much less cpu (1700% down to 240%).  No new data has been added to the collection.

**Comment by hawkphantomnet** [comment]
![top_low_cpu](https://user-images.githubusercontent.com/13686563/101621776-2b691100-3a51-11eb-98b5-a8bb3d57f040.png)
above is the top details for the threads of milvus_server, querying vectors with index before I restart the milvus container. 

![top_high_cpu](https://user-images.githubusercontent.com/13686563/101621923-5a7f8280-3a51-11eb-901e-6fa9f25a3a4b.png)
the top details after i restart milvus container and perform the same query

**Comment by hawkphantomnet** [comment]
[perf.data_high_cpu.zip](https://github.com/milvus-io/milvus/files/5665870/perf.data_high_cpu.zip)
[perf.data_low_cpu.zip](https://github.com/milvus-io/milvus/files/5665871/perf.data_low_cpu.zip)

**Comment by hawkphantomnet** [comment]
comparing the perf.data above, the cpu usage increased was increased in calling libgomp
![perf_data_low_cpu](https://user-images.githubusercontent.com/13686563/101631575-86a20000-3a5f-11eb-9660-2ee1bdc01bc6.png)
![perf_data_high_cpu](https://user-images.githubusercontent.com/13686563/101631588-899cf080-3a5f-11eb-920f-f1423fa54e82.png)

**Comment by hawkphantomnet** [comment]
i have tried the config below but not helping: 
```
engine_config:
  use_blas_threshold: 800
  omp_thread_num: 1

**Comment by hawkphantomnet** [comment, marked as answer]
setting environment variable OMP_WAIT_POLICY=PASSIVE fix this issue

**Comment by yhmo** [comment]
@hawkphantomnet 
It could relate to NUMA. Please refer to this issue: #4208
